subsys/bluetooth/host/hci_core.c

```diff
@@ -1032,10 +1032,20 @@ static void enh_conn_complete(struct bt_hci_evt_le_enh_conn_complete *evt)
 	struct bt_conn *conn;
 	int err;
 
-	BT_DBG("status 0x%02x handle %u role %u %s", evt->status, handle,
-	       evt->role, bt_addr_le_str(&evt->peer_addr));
+	BT_DBG("status 0x%02x handle %u role %u adr:%s rpa:%s local_rpa:%s",
+			evt->status, handle, evt->role, bt_addr_le_str(&evt->peer_addr),
+			bt_addr_str(&evt->peer_rpa), bt_addr_str(&evt->local_rpa));
 
 #if defined(CONFIG_BT_SMP)
+	if (!bt_addr_cmp(&evt->peer_addr.a, BT_ADDR_NONE)) {
+		if (bt_addr_cmp(&evt->peer_rpa, BT_ADDR_NONE)) {
+			BT_DBG("RPA is ok. Copy to peer address");
+			bt_addr_copy(&evt->peer_addr.a, &evt->peer_rpa);
+			evt->peer_addr.type = BT_ADDR_LE_RANDOM;
+		} else {
+			BT_WARN("Invalid peer address %s", bt_addr_le_str(&evt->peer_addr));
+		}
+	}
```

Comment:
minor coding style: empty line after this (whenever the next line is not also a `}`)